src/integration-test/java/com/mongodb/jdbc/integration/MongoIntegrationTest.java

```diff
@@ -0,0 +1,134 @@
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import org.junit.jupiter.api.Test;
+
+public abstract class MongoIntegrationTest {
+
+    /**
+     * Creates a new connection.
+     *
+     * @param extraProps Extra properties on top of the default ones that the class implementating
+     *     it is using.
+     * @return The connection.
+     * @throws SQLException If the connection can not be created.
+     */
+    public abstract Connection getBasicConnection(Properties extraProps) throws SQLException;
+
+    /** Simple callable used to spawn a new statement and execute a query. */
+    public class SimpleQueryExecutor implements Callable<Void> {
+        private final Connection conn;
+        private final String query;
+
+        public SimpleQueryExecutor(Connection conn, String query) {
+            this.conn = conn;
+            this.query = query;
+        }
+
+        @Override
+        public Void call() throws Exception {
+            try (Statement stmt = conn.createStatement()) {
+                stmt.executeQuery(query);
+            }
+            return null;
+        }
+    }
+
+    /**
+     * Verifies that concurrent connections and statements are not impacting the logging
+     * capabilities. Concurrent connection must be able to create log files with no issues.
+     * Concurrent statements must be able to write their messages in the same log file.
+     */
+    @Test
+    public void testLoggingWithParallelConnectionAndStatementExec() throws Exception {
+        ExecutorService executor = Executors.newFixedThreadPool(4);
+        List<Callable<Void>> tasks = new ArrayList<Callable<Void>>();
+
+        // Connection with no logging.
+        Connection noLogging = connect(null);
+        // Connection only logging exceptions.
+        Connection logErrorOnly = connect(Level.SEVERE);
+        // Connection logging all public method entries of the JDBC interface.
+        Connection logEntries = connect(Level.FINER);
+        try {
+            addSimpleQueryExecTasks(tasks, noLogging);
+            addSimpleQueryExecTasks(tasks, logErrorOnly);
+            addSimpleQueryExecTasks(tasks, logEntries);
+            executor.invokeAll(tasks);
+        } finally {
+            executor.awaitTermination(1, TimeUnit.SECONDS);
+            if (noLogging != null) {
+                cleanUp((MongoConnection) noLogging);
+            }
+            if (logErrorOnly != null) {
+                cleanUp((MongoConnection) logErrorOnly);
+            }
+            if (logEntries != null) {
+                cleanUp((MongoConnection) logEntries);
+            }
+        }
+    }
+
+    /**
+     * Connect with the given logging level.
+     *
+     * @param logLevel The log level or null if not logging.
+     * @return the connection.
+     * @throws SQLException If an error occurs during the connection process.
+     */
+    private Connection connect(Level logLevel) throws SQLException {
```

Comment:
If I understand this test correctly, with the latest changes we now have to set a LogDir  for the logger to use to create the log file in.   Here without a directory it is not creating the connection_*.log file to test that concurrent file creation works.

Reply:
Oh, you are right. I made the changes for the unit tests but not for integration tests.
Fixed. It will log in the current directory.